### PR TITLE
chore: log sharky recovery

### PR DESCRIPTION
**Pull request by istae**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Screenshots (if appropriate):

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3487)

### CODE REVIEW

File: pkg/localstore/localstore.go [hunk 1]
```diff
@@ -670,6 +629,50 @@ func New(path string, baseKey []byte, ss storage.StateStorer, o *Options, logger
...
 
+func (db *DB) safeInit(rootPath, sharkyBasePath string) error {
+	// create if needed
+	path := filepath.Join(rootPath, sharkyDirtyFileName)
+	if _, err := os.Stat(path); errors.Is(err, fs.ErrNotExist) {
+		// missing lock file implies a clean exit then create the file and return
+		return os.WriteFile(path, []byte{}, 0644)
+	}
+
+	db.logger.Warning("localstore sharky .DIRTY file exists: starting recovery due to previous dirty exit")
```
Comment (ldeffenb) [inline pkg/localstore/localstore.go:640]: Awesome!   You can't imagine how many times I've sat and wondered why my node was taking so long to establish peers.  I've always suspected it was due to an unclean shutdown, but hopefully this log will help me to confirm or deny my theory.

File: pkg/localstore/localstore.go [hunk 2]
```diff
@@ -670,6 +629,50 @@ func New(path string, baseKey []byte, ss storage.StateStorer, o *Options, logger
...
+	if err != nil {
+		return err
+	}
+
+	err = recoverySharky.Close()
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (ldeffenb) [inline pkg/localstore/localstore.go:673]: Any chance of getting a "done" message, possibly with the elapsed time when it finishes?   Either here or maybe on a defer and show the error or success from there?

Comment (istae) [reply pkg/localstore/localstore.go:677]: sure, check the PR again :)

File: pkg/localstore/localstore.go (hunk 1 above)
Comment (istae) [reply pkg/localstore/localstore.go]: so after this change, can you report how often the recovery happens to you if at all ?

File: pkg/localstore/localstore.go [hunk 3]
```diff
@@ -637,7 +637,11 @@ func (db *DB) safeInit(rootPath, sharkyBasePath string) error {
 		return os.WriteFile(path, []byte{}, 0644)
 	}
 
-	db.logger.Warning("localstore sharky .DIRTY file exists: starting recovery due to previous dirty exit")
+	defer func(t time.Time) {
```
Comment (ldeffenb) [inline pkg/localstore/localstore.go:640]: Doubly awesome!  Thanks!

File: pkg/localstore/localstore.go (hunk 1 above)
Comment (ldeffenb) [reply pkg/localstore/localstore.go]: I'll try to remember to keep an eye on it.   Normally, I've only observed startup delays after a shutdown doesn't go cleanly, either an OOM kill on Ubuntu, or a Control-C on Windows that eventually reports that something terminated with running goroutine.  Or when I get impatient and just hit double Control-C's on windows which instantly kills the process with no hope of any shutdown cleanup!

**Review by notanatol — APPROVED** [review]

**Review by aloknerurkar — APPROVED** [review]

**Pull request merged**